openupgrade_scripts/scripts/stock_account/17.0.1.1/post-migration.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 Viindoo Technology Joint Stock Company (Viindoo)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def _stock_move_convert_analytic_account_line_id_m2o_to_m2m(env):
+    openupgrade.m2o_to_x2m(
+        env.cr,
+        env["stock.move"],
+        "stock_move",
+        "analytic_account_line_ids",
+        "analytic_account_line_id",
+    )
+
+
+def fill_account_move_line_cogs_origin_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move_line aml
+        SET cogs_origin_id = line.id
+        FROM account_move am
+        JOIN account_move_line line ON line.move_id = am.id
+                                    AND line.display_type != 'cogs'
+        WHERE aml.move_id = am.id AND aml.display_type = 'cogs'
```

Comment:
```suggestion
        FROM account_move_line line
        WHERE aml.move_id = line.move_id AND line.display_type != 'cogs' AND aml.display_type = 'cogs'
```

(we don't need the join over account_move)